electrumsv/tests/test_wallet.py

```diff
@@ -550,64 +619,169 @@ def test_legacy_wallet_loading(mock_wallet_app_state, storage_info: WalletStorag
         for public_key, encrypted_prv in private_key_keystore._keypairs.items():
-            prv_keypairs[public_key.to_hex()] = pw_decode(encrypted_prv, initial_password)
-
-    password = "654321"
-    future, update_completion_event = wallet.update_password(initial_password, password)
+            keyinstance_private_keys[public_key.to_hex()] = pw_decode(encrypted_prv,
+                initial_password)
+            test_state.count_of_keyinstance_privkey_reencryptions += 1
+
+    # Derivation data in the MasterKeys table.
+    master_key_xprvs: list[str] = []
+    master_key_seeds: list[str] = []
+    master_key_passphrases: list[str] = []
+    for keystore in wallet.get_keystores():
+        data = keystore.to_derivation_data()
+        for entry_name in ("seed", "passphrase", "xprv"):
+            entry_value = cast(str | None, data.get(entry_name, None))
+            if not entry_value:
+                continue
+            decoded_value = pw_decode(entry_value, initial_password)
+            if entry_name == "seed":
+                master_key_seeds.append(decoded_value)
+                test_state.count_of_masterkeys_table_seed_reencryptions += 1
+            if entry_name == "xprv":
+                master_key_xprvs.append(decoded_value)
+                test_state.count_of_masterkeys_table_xprv_reencryptions += 1
+            if entry_name == "passphrase":
+                master_key_passphrases.append(decoded_value)
+                test_state.count_of_masterkeys_table_passphrase_reencryptions += 1
+
+    # Newer wallets have a `Servers` table with an `encrypted_api_key` column
+    encrypted_api_keys = list[str]()
+    for server in wallet.data.read_network_servers():
+        if server.encrypted_api_key is not None:
+            test_state.count_of_server_api_key_reencryptions += 1
+            decoded_value = pw_decode(server.encrypted_api_key, initial_password)
+            encrypted_api_keys.append(decoded_value)
+
+    new_password = "654321"
+    future, update_completion_event = wallet.update_password(initial_password, new_password)
     # Wait for the database update to finish.
     future.result(5)
     # Wait for the done callback to finish.
     update_completion_event.wait()
+    wallet.stop()
+
+    mock_wallet_app_state.credentials.get_wallet_password = lambda wallet_path: new_password
+    # Perform a full shutdown and reload of the wallet database. This will cover the code
+    # pathways of:
+    # - flushing the network server cache updates to the database. This was causing a bug where
+    # it was overwriting the updated encrypted api key with the old version.
+    # - reloading of the `WalletData` table variables into the wallet cache (see:
+    # `Wallet._load_servers()`. This involves json deserialization of stored values)
+    # - loading the `Server` table data and decrypting the api keys with the new password
+    storage = WalletStorage(wallet_path)
```

Comment:
Please remove this unnecessary wallet reloading. It is testing a whole lot of things we do not care about and not testing the thing we wanted in the first place. 

For each case where there is encrypted data in any encountered wallet, after the password change look at the in-memory cached version and verify that it was updated correctly. Bump the count for the found checked encrypted data type then to indicate it was successfully updated.